common/djangoapps/student/cookies.py

```diff
@@ -118,6 +118,17 @@ def set_user_info_cookie(response, request):
     )
 
 
+def set_is_enterprise_cookie(request, response, is_enterprise):
+    """ Sets the is_enterprise cookie on the response. """
+    cookie_settings = standard_cookie_settings(request)
+    response.set_cookie(
+        'is_enterprise',
+        json.dumps(is_enterprise),
+        secure=None,
+        **cookie_settings
+    )
```

Comment:
Will it be possible for any client to make a request to ecommerce with this cookie and be identified as an enterprise user? My concern is that there is some logic in ecommerce related to automated discounts for enterprise learners that would trigger a discount for any client that sends this cookie.

Reply:
i'm not sure what you mean
i added a new cookie as far as I'm aware that is not currently used anywhere (unless the name is colliding with an existing cookie?)
i was going to use this cookie for future optimizely tests by excluding users who have this cookie from the test

Reply:
Right, but could the end user edit this cookie to impersonate an enterprise learner?

Reply:
If that is possible, then I'm afraid that an engineer would find this cookie and assume it is the primary way for determining whether or not a user is related to an enterprise.

Reply:
i'm not sure i understand what you're asking.
are you wondering if someone in the future will find this code in the lms, write new code in ecommerce to apply discounts based on this cookie, and then users will add a cookie to get discounts?
it seems like applying discounts based on client side data is something we will want to avoid regardless

or are you concerned people will use this cookie for something other than optimizely tests?
we could rename the cookie to be optimizely specific but is it not true that all enterprise users would have this cookie based on the function that populates it? or is this not correct

Reply:
Yes, those are both concerns I have. I like the idea of renaming the cookie to indicate the specific purpose of it. Probably a comment in the code as well would be good.

Reply:
One other thing you might consider doing is to store the name of the `EnterpriseCustomer` which the user is linked to instead of a boolean. This would allow you to filter experiments for specific EnterpriseCustomers should the need arise.

Reply:
can you clarify why you're concerned that someone might use this to determine if a user is enterprise? i don't want to check this way if you think this approach will give wrong results

Reply:
The scenario I'm envisioning is a malicious user edits the value of the cookie to gain inappropriate access to resources on our side. I think renaming the cookie and explicitly stating in the code that it should not be used for mission critical workflows to determine if a user is linked to an enterprise covers that concern.

Reply:
i see, i guess i assumed not using client-side data for critical payment related things was self-evident, but maybe not
how is this name?
is_enterprise_notforpayments

Reply:
I would probably go with something like `experiments_is_enterprise`, but you should decide the name. Actually, I think the more important part is the comment in the code identifying what you are using the flag for.

Reply:
i'll add a comment regarding this as well.
ok i'll go with experiments_is_enterprise, though there may be non-critical things not related to experiments that this would be fine to use for, our team at least won't care about those cases